account_due_list_payment_mode_edit/models/account_move.py

```diff
@@ -0,0 +1,58 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    payments_count = fields.Integer(string="Payments Count",
+                                    compute='_compute_payments_count',
+                                    readonly=True)
+
+    def _compute_payments_count(self):
+        for move in self:
+            if move.is_invoice():
+                payments = move.line_ids.filtered(
+                    lambda r: r.account_type in (
+                        "asset_receivable",
+                        "liability_payable",
+                    ))
+                move.payments_count = len(payments)
+            else:
+                move.payments_count = 0
+
+    def action_post(self):
+        res = super(AccountMove, self).action_post()
+        for move in self:
+            if move.is_invoice():
+                move.line_ids._compute_payment_mode()
+        return res
+
+    def action_view_payments(self):
+        self.ensure_one()
+        payments = self.line_ids.filtered(
+            lambda r: r.account_type in (
+                "asset_receivable",
+                "liability_payable",
+            ))
+        action = self.env.ref(
+            'account_due_list.action_invoice_payments').read()[0]
```

Comment:
`read()` for actions method will fire an error for users without full Settings permissions, you can check. Use `_for_xml_id` method instead and adapt code to it